### ISSUE TITLE
Final Projects - TypeWriter

### ISSUE THREAD

**Issue by zs-sz** [zs-sz/METU---Generative-Design-101#3]

**Comment by barizsidek** [comment]
Hello, 
Mennyire kell customizálni a file-t? Nekem itt megállt a kreatívitásom jelenleg. Illetve hova küldjem el a "kész" kódot? Köszönöm, Barbara

![screen shot 2017-05-22 at 16 28 05](https://cloud.githubusercontent.com/assets/28146475/26313634/c3958034-3f0b-11e7-9826-31c34a8a0ebd.png)

**Comment by zs-sz** [comment]
csatold ide a foldert becsomagolva. este megnezem.

**Comment by zs-sz** [comment]
@barizsidek 
csatold ide a foldert becsomagolva. este megnezem.

**Comment by barizsidek** [comment]
Oké! Ha azt mondod, hogy még írjak hozzá, akkor írok! 

Illetve egy kérdésem lenne... van bármi "egyszerű" módja, hogy egy forma körvonalát szagatottá tegyek? A css-ben annyira egyszerű, itt viszont csak olyan módszereket találtam, amik nem működnek.
Köszönöm!  

[TypeWriter 2.zip](https://github.com/zs-sz/METU---Generative-Design-101/files/1019169/TypeWriter.2.zip)

**Comment by zs-sz** [comment]
hat igy hirtelen nem tudom, de megnezem ezt is kesobb. valoszinuleg lehet.

**Comment by zs-sz** [comment]
@barizsidek 
Na a szaggatott vonal lehetseges, de tul korulmenyes ebben a formaban. 

viszont:
Hasznald a kovetkezo szoveget: 
"Generative design is a form finding process that can mimic natures evolutionary approach to design."
(vagy hasznalsz sajatot de legyen valami ertelme es hasonlo hosszusagu.

Lehetoleg valami kevesbe 'display' betutipussal. Amit minimum hozza kene adnod:
- Variald az egyes karakterek szineit vagy veletlenszeruen vagy fixen az adott betuhoz kepest.
- Magat a geometriat is modosithatod mindenfele veletlenszeru alakzatokkal csal maradjon olvashato. 
- Ments le 4-5 verziot es kuld el

Tuti otos: 
- ha egyes irasjelek helyere egyedi grafikat csinalsz. peldaul szokoz, vesszo, stb lenyomasanal mas grafika jelenik meg

Holnap meg egesz nap elerheto leszek, illetve ma este is. Ha elakadsz szolj.

**Comment by barizsidek** [comment]
bocs, de teljesen elakadtam... 

kiolvastam az interneten mindent arról, hogy miként lehetne megváltoztatni billentyű-lenyomással a karakterek színeit, de semmi se működik a kóddal.... ? 
a geometria szintén.nagyon tanácstalan vagyok, hogy akkor most hogyan, nézzegettem más kódokat is, meg azon órák kódját amik meg vannak. esetleg van valami oldal, amit tudsz ajánlani? a google keresés nálam már nagyon kifutott és tényleg nem tudom mit lehetne még..
Köszi!

az egyedi írásjeleket simán beletudtam tenni...

**Comment by zs-sz** [comment]
@barizsidek 
a karakter maga a glyph.js-ben van. 
[link](https://github.com/zs-sz/METU---Generative-Design-101/blob/master/FinalProjectTemplates/TypeWriter/js/glyph.js
)
eloszor a hatteret rajzloja ki utana magat a betut (lasd a kommenteket)
namost ha ide beszursz egy olyat peldaul hogy: 

`
// YOUR DESIGN COMES HERE
// draw the background here
// ha a betu a VAGY(||) A akkor legyen fekete a hatter minden mas esetben legyen feher

	if (this.letter == 'a' || this.letter == 'A') {
		fill(0)
	} else {
		fill(255)
	}
	rect(
		this.x,
		this.y,
		this.size,
		this.size
	);
`

**Comment by barizsidek** [comment]
Köszi! Eddig is a glyph.js-ben próbálkoztam - de  bent maradt egy sor a sketch-ben is... Megnéztem ezzel a kóddal - nem működött, ekkor lett nagyon gyanús a dolog, átolvastam mindent a két fájlban és sikerült rájönnöm, mi volt a hiba! Köszönöm. :) 

Az egyedi grafikánál kép legyen vagy szimbólum?

**Comment by zs-sz** [comment]
az egyedi grafikanal csak arra gondoltam, hogy valamivel komplexebb dolgot rajzolsz mint a sima rect fuggveny. Kombinalod a tobbi rajzolo fuggvennyel(ellipse, triangle, line, stb). Ilyenkor arra figyelj hogy valoszinuleg a rajzolas is az if() v. else() agba kerul. Az elozo peldaban csak a fillt tettem oda. Nyilvan arra ugyelj hogy barmit is rajzolsz, lehetoleg ne legyen nagyobb mint a this.size alatt beallitott meret. 

Illetve olyat is csinalhatsz hogy az if-ben nem a letter szamit hanem mondjuk random szam egy es ot kozott es akkor van ot fele hattered. 

var chance = round(random(1,5))
if(chance == 1) {
  // rajzol,rajzol
} else if (chance == 2) {
 // rajzol, rajzol
} else {
// rajzol, rajzol
}

..de amit jonak latsz.

**Comment by barizsidek** [comment]
hello, jelenleg itt tartok - ez így jó lesz? :) Köszönöm a segítséget! 

![screen shot 2017-05-24 at 22 02 11](https://cloud.githubusercontent.com/assets/28146475/26423227/1e8fb65e-40cd-11e7-80ea-46bed188d503.png)
![screen shot 2017-05-24 at 22 02 33](https://cloud.githubusercontent.com/assets/28146475/26423228/1e9683e4-40cd-11e7-9dfa-ee645ec7a723.png)
![screen shot 2017-05-24 at 22 04 10](https://cloud.githubusercontent.com/assets/28146475/26423235/227ab76e-40cd-11e7-932a-9116e9f8ae8f.png)
![screen shot 2017-05-24 at 22 04 48](https://cloud.githubusercontent.com/assets/28146475/26423238/23e5a8de-40cd-11e7-9b9f-e18cb22ec95b.png)

**Comment by zs-sz** [comment]
igen. kuld majd at a kodot. csatolhatod itt is egy kommentben. reggel megnezem, de a kepek alapjan jo lesz.

**Comment by barizsidek** [comment]
[TypeWriter 3.zip](https://github.com/zs-sz/METU---Generative-Design-101/files/1027164/TypeWriter.3.zip)
